### PR TITLE
add user profile widget RELEASE

### PR DESCRIPTION
**Pull request by nirgur**
## Related Issues

Fixes <link_to_github_issue>

## Related PRs

| branch       | PR         |
| ------------ | ---------- |
| service a PR | Link to PR |
| service b PR | Link to PR |

## Description

A few sentences describing the overall goals of the pull request's commits.

## Must

- [ ] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -511,6 +511,26 @@ The `AuditManagement` widget will let you embed an audit table in your site.
...
+
+- Update user profile picture
+- Update user personal information
+- Update authentication methods
+- Logout
+
+###### Usage
+
+```html
+<user-profile widgetId="user-profile-widget" />
```
Comment (dorsha) [inline README.md:528]: Please add logout callback

File: package.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
 		"@descope/audit-management-widget": "0.1.0",
 		"@descope/role-management-widget": "0.1.35",
 		"@descope/user-management-widget": "0.4.36",
+		"@descope/user-profile-widget": "0.0.15",
```
Comment (dorsha) [inline package.json:40]: Update all other widgets and web-component

File: projects/angular-sdk/src/lib/components/user-profile/user-profile.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		this.setupWebComponent();
+		this.elementRef.nativeElement.appendChild(this.webComponent);
+	}
+
+	ngOnChanges(): void {
+		this.setupWebComponent();
+	}
+
+	private setupWebComponent() {
+		this.webComponent.setAttribute('project-id', this.projectId);
```
Comment (dorsha) [inline projects/angular-sdk/src/lib/components/user-profile/user-profile.component.ts:40]: Don't we need to have here the logout callback?

File: projects/demo-app/src/app/my-user-profile/my-user-profile.component.html [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<user-profile
+	[theme]="theme"
+	[debug]="debugMode"
+	widgetId="audit-management-widget"
```
Comment (dorsha) [inline projects/demo-app/src/app/my-user-profile/my-user-profile.component.html:4]: Add here the logout callback

File: README.md (hunk 1 above)
Comment (nirgur) [reply README.md]: done

File: projects/angular-sdk/src/lib/components/user-profile/user-profile.component.ts (hunk 3 above)
Comment (nirgur) [reply projects/angular-sdk/src/lib/components/user-profile/user-profile.component.ts:50]: done

File: package.json (hunk 2 above)
Comment (nirgur) [reply package.json:40]: done

File: README.md [hunk 5]
```diff
@@ -511,6 +511,28 @@ The `AuditManagement` widget will let you embed an audit table in your site.
...
+- Update user profile picture
+- Update user personal information
+- Update authentication methods
+- Logout
+
+###### Usage
+
+```angular2html
+<user-profile widgetId="user-profile-widget"
+(logout)="<your_logout_function>"
```
Comment (dorsha) [inline README.md:529]: Change to onLogout

File: projects/angular-sdk/src/lib/components/user-profile/user-profile.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+				DescopeAuthConfig,
+				{ provide: DescopeAuthConfig, useValue: mockConfig }
+			]
+		});
+
+		fixture = TestBed.createComponent(UserProfileComponent);
+		component = fixture.componentInstance;
+		component.projectId = '123';
+		component.widgetId = 'widget-1';
+		component.logout = new EventEmitter<CustomEvent>();
```
Comment (dorsha) [inline projects/angular-sdk/src/lib/components/user-profile/user-profile.component.spec.ts:53]: onLogout

File: projects/angular-sdk/src/lib/components/user-profile/user-profile.component.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+		}
+		if (this.debug) {
+			this.webComponent.setAttribute('debug', this.debug.toString());
+		}
+
+		if (this.logger) {
+			(this.webComponent as any).logger = this.logger;
+		}
+
+		if (this.logout) {
```
Comment (dorsha) [inline projects/angular-sdk/src/lib/components/user-profile/user-profile.component.ts:66]: onLogout..

File: projects/demo-app/src/app/my-user-profile/my-user-profile.component.html [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+	(logout)="logout($event)"
```
Comment (dorsha) [inline projects/demo-app/src/app/my-user-profile/my-user-profile.component.html:5]: onLogout

**Review by dorsha — APPROVED** [review]

**Pull request merged**